static/src/js/components/RevisionList/RevisionList.jsx

```diff
@@ -0,0 +1,251 @@
+    },
+    {
+      dataKey: 'revisionDate',
+      title: 'Revision Date',
+      className: 'col-auto',
+      dataAccessorFn: (cellData) => cellData.split('T')[0]
+    },
+    {
+      dataKey: 'userId',
+      title: 'Action by',
+      className: 'col-auto'
+    },
+    {
+      // Change dataKey to 'revisionId' then fix unique key issue (see line 116)
+      dataKey: 'actions',
+      title: 'Actions',
+      className: 'col-auto',
+      dataAccessorFn: buildActionCell
+    }
+  ]
+
+  const [columns, setColumns] = useState(getColumnState())
+
+  useEffect(() => {
+    setColumns(getColumnState())
+  }, [type])
+
+  const pageTitle = loading && items.length === 0
+    ? `Loading ${startCase(getHumanizedNameFromTypeParam(type))} Revisions`
+    : `${commafy(count)} ${startCase(getHumanizedNameFromTypeParam(type))} ${pluralize('Revisions', count)}`
+
+  return (
+    <Page
+      pageType="secondary"
+      title={pageTitle}
+      breadcrumbs={
+        [
+          {
+            label: 'Revision Results',
+            active: true
+          }
+        ]
+      }
+    >
+      <Row>
+        <Col sm={12}>
+          {error && <ErrorBanner message={parseError(error)} />}
+          {
+            !error && (
+              <ControlledPaginatedContent
+                activePage={activePage}
+                count={count}
+                limit={limit}
+                setPage={setPage}
+              >
+                {
+                  ({
+                    totalPages,
+                    pagination,
+                    firstResultPosition,
+                    lastResultPosition
+                  }) => {
+                    const paginationMessage = `Showing ${totalPages > 1 ? `${firstResultPosition}-${lastResultPosition} of` : ''} ${count} `
+                        + `${pluralize('Revision', count)}`
+
+                    return (
+                      <>
+                        <Row className="d-flex justify-content-between align-items-center mb-4">
+                          <Col className="flex-grow-1" xs="auto">
+                            {
+                              !count && loading && (
+                                <div className="w-100">
+                                  <span className="d-block">
+                                    <Placeholder as="span" animation="glow">
+                                      <Placeholder xs={8} />
+                                    </Placeholder>
+                                  </span>
+                                </div>
+                              )
+                            }
+                            {
+                              (count || (!loading && !count)) && (
+                                <span className="text-secondary fw-bolder">{paginationMessage}</span>
+                              )
+                            }
+                          </Col>
+                          {
+                            totalPages > 1 && (
```

Comment:
While the pagination component could include this logic, I'm not sure if it makes sense to keep it out of the parent component. If we took out the logic and just did {pagination}, then in the event of totalPages !> 1 it would render an empty <Col> component, correct?